solver/jobs.go

```diff
@@ -933,7 +934,10 @@ func notifyStarted(ctx context.Context, v *client.Vertex, cached bool) {
 	v.Started = &now
 	v.Completed = nil
 	v.Cached = cached
-	pw.Write(v.Digest.String(), *v)
+	// id dedupes progress updates; use of the start time ensures that distinct
+	// stop/start intervals of the vertex don't get deduped with one another.
+	id := v.Digest.String() + strconv.Itoa(int(v.Started.UnixNano()))
```

Comment:
I'm not sure about using started time there. Might as well just use random then. If this is for the second cache key then what if we just pass a suffix for that case. Then at least the caller side still controls the ID. Could even be variadic I think.